src/components/Auth/ForgotPassword/ForgotPassword.react.js

```diff
@@ -1,234 +1,132 @@
+      this.setState({
+        email,
+        emailErrorMessage: emailError ? <Translate text="Invalid Email" /> : '',
+      });
     }
-    this.setState(state);
   };
 
-  // Submit the Forgot Password Form
   handleSubmit = event => {
     event.preventDefault();
 
-    let email = this.state.email.trim();
-    let validEmail = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email);
-
-    let defaults = UserPreferencesStore.getPreferences();
-    let BASE_URL = defaults.Server;
+    const { actions } = this.props;
+    const { email, emailErrorMessage } = this.state;
 
     this.setState({ loading: true });
 
-    let serverUrl = this.state.serverUrl;
-    if (serverUrl.slice(-1) === '/') {
-      serverUrl = serverUrl.slice(0, -1);
-    }
-    if (serverUrl !== '') {
-      BASE_URL = serverUrl;
-    }
-    if (email && validEmail) {
-      $.ajax({
-        url: BASE_URL + '/aaa/recoverpassword.json?forgotemail=' + email,
-        dataType: 'jsonp',
-        crossDomain: true,
-        timeout: 3000,
-        async: false,
-        statusCode: {
-          422: function() {
-            let msg = 'Email does not exist';
-            let state = this.state;
-            state.msg = msg;
-            state.loading = false;
-            this.setState(state);
-          },
-        },
-        success: function(response) {
-          let msg = response.message;
-          let state = this.state;
-          state.msg = msg;
-          if (response.accepted) {
-            state.success = true;
-          } else {
-            state.success = false;
-            state.msg += 'Please Try Again';
-          }
-          state.loading = false;
-          this.setState(state);
-        }.bind(this),
-        error: function(jqXHR, textStatus, errorThrown) {
-          let jsonValue = jqXHR.status;
-          let msg = '';
-          if (jsonValue === 404) {
-            msg = 'Email does not exist';
+    if (email && !emailErrorMessage) {
+      this.setState({ loading: true });
+      actions
+        .getForgotPassoword({ email })
+        .then(({ payload }) => {
+          let dialogMessage = payload.message;
+          let success;
+          if (payload.accepted) {
+            success = true;
           } else {
-            msg = 'Failed. Try Again';
+            success = false;
+            dialogMessage = 'Please Try Again';
           }
-          if (status === 'timeout') {
-            msg = 'Please check your internet connection';
-          }
-          let state = this.state;
-          state.msg = msg;
-          state.loading = false;
-          this.setState(state);
-        }.bind(this),
-      });
+          this.setState({
+            success,
+            dialogMessage,
+            loading: false,
+          });
+        })
+        .catch(error => {
+          this.setState({
+            loading: false,
+            dialogMessage: 'Failed. Try Again',
+          });
+        });
     }
```

Comment:
Make loading false here

Reply:
I have shifted {loading: true} inside the if block. So I guess {loading: false} outside if block will not be required any more